### PR TITLE
database performance audit

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
[Project requirements (1h)](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_performance_audit.md#project-requirements-1h)
The following queries are taking too much time (1 sec = 1000ms can be considered as too much time for database query). Try them on your machine to confirm it:

SELECT COUNT(*) FROM visits where animal_id = 4;
SELECT * FROM visits where vet_id = 2;
SELECT * FROM owners where email = 'owner_18327@mail.com';
Use EXPLAIN ANALYZE on the previous queries to check what is happening. Take screenshots of them - they will be necessary later.

Find a way to decrease the execution time of the first query. Look for hints in the previous lessons.

Find a way to improve execution time of the other two queries.

While you are making changes, check if they help by running EXPLAIN ANALYZE. Once you succeed, take a screenshot of the EXPLAIN ANALYZE result showing that time actually decreased.

Any changes you made to your database schema should be added to the schema.sql file and commited in a new pull request.

If you decide to commit any INSERT INTO queries - remember to add the m to the data.sql file. It is important to keep only the queries that change the database structure in the schema.sql file.

In your pull request description include screenshots with EXPLAIN ANALYZE results (before and after you improve the speed) for each of the 3 problematic queries.

NOTE: once you are done with this task feel free to remove your database in order to free up some space on your computer. With database schema you will be able to re-create it at any point of time.
query1BeforeUpdate
query1After
query2Before
query2After
query3BeforeUpdate
query3After

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -2,192 +2,306 @@
...
+EXPLAIN ANALYZE
+SELECT *
+FROM visits
+where vet_id = 2;
+
+-- Expplain and analize
+EXPLAIN ANALYZE
+SELECT *
+FROM owners
+where email = 'owner_18327@mail.com';
```
Comment (Gambit142) [inline queries.sql]: - [ ] The execution time for the two queries highlighted below is below **1000ms (1 second)**. 

```sql
EXPLAIN ANALYZE
SELECT COUNT(*)
FROM visits
where animal_id = 4;

EXPLAIN ANALYZE
SELECT *
FROM owners
where email = 'owner_18327@mail.com';
```

According to the requirements, the execution time of each of these queries should be above **1000ms (1 second)**. I would advise you to run these queries one or two more times to ensure there are many entries in the table before you run the query 👍 😎 :

```sql
INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
```

#### Hint: Please update your screenshots after implementing this. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Abdulwhbhussain @Errytagedesign 👋, 

Good job so far!

## Highlights
- Added necessary columns ✔️ 
- Created data for tables ✔️ 
- Wrote the required queries ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: queries.sql (hunk 1 above)
Comment (Abdulwhbhussain) [reply queries.sql]: Implemented!

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @Abdulwhbhussain , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**